src/main/java/ui/components/pickers/LabelPickerState.java

```diff
@@ -0,0 +1,271 @@
+package ui.components.pickers;
+
+import backend.resource.TurboLabel;
+import util.Utility;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+public class LabelPickerState {
+    List<String> addedLabels;
+    List<String> removedLabels;
+    Set<String> initialLabels;
+    List<String> matchedLabels;
+    OptionalInt currentSuggestionIndex;
+
+    private LabelPickerState(Set<String> initialLabels,
+                             List<String> addedLabels,
+                             List<String> removedLabels,
+                             List<String> matchedLabels,
+                             OptionalInt currentSuggestionIndex) {
+        this.initialLabels = initialLabels;
+        this.addedLabels = addedLabels;
+        this.removedLabels = removedLabels;
+        this.matchedLabels = matchedLabels;
+        this.currentSuggestionIndex = currentSuggestionIndex;
+    }
+
+    private LabelPickerState(Set<String> initialLabels,
+                             List<String> addedLabels,
+                             List<String> removedLabels,
+                             List<String> matchedLabels) {
+        this.initialLabels = initialLabels;
+        this.addedLabels = addedLabels;
+        this.removedLabels = removedLabels;
+        this.matchedLabels = matchedLabels;
+        this.currentSuggestionIndex = OptionalInt.empty();
+    }
+
+    /*
+     * Methods for Logic
+     */
+
+    public LabelPickerState(Set<String> initialLabels) {
+        this.initialLabels = initialLabels;
+        this.addedLabels = new ArrayList<>();
+        this.removedLabels = new ArrayList<>();
+        this.matchedLabels = new ArrayList<>();
+        this.currentSuggestionIndex = OptionalInt.empty();
+    }
+
+    public LabelPickerState clearMatchedLabels() {
```

Comment:
This doesn't sound right. Method sounds like a mutator but acts like a factory method.